src/openseadragon.js

```diff
@@ -816,6 +816,32 @@ window.OpenSeadragon = window.OpenSeadragon || function( options ){
                     canvasElement.getContext( '2d' ) );
     }());
 
+    /**
+     * @private
+     * @inner
+     * @function
+     * @returns {Number} The device's pixel density ratio, or null if canvas isn't supported by the browser.
+     */
+    $.pixelDensityRatio = (function () {
+        if ( $.supportsCanvas ) {
```

Comment:
Since it looks like this function will be used a good deal, we should calculate the value once and then return the cached value every time after that.